packages/webapp/pages/join.tsx

```diff
@@ -0,0 +1,236 @@
+                </FlexCentered>
+              </Button>
+            </FlexCol>
+            <FlexCol className="relative mt-8 laptop:mt-0">
+              <button
+                className="-mx-8"
+                type="button"
+                onClick={() => {
+                  setVideoOpen(true);
+                }}
+              >
+                <Image
+                  className="object-contain w-full laptopL:max-w-[1015px] laptopL:height-[633px] laptop:max-w-[650px] height-[405px]"
+                  src={cloudinary.referralCampaign.appScreenshot}
+                  alt="Image showing the daily.dev"
+                />
+                <PlayIcon
+                  style={{
+                    boxShadow: '0px 3px 43px rgba(206,60,243, 1)',
+                  }}
+                  className="absolute top-1 laptop:top-1/2 left-1/2 z-1 h-20 laptop:h-28 bg-pepper-90 rounded-full transform -translate-x-1/2 -translate-y-1/2 !w-20 laptop:!w-28"
+                  size={IconSize.XXXLarge}
+                  secondary
+                />
+              </button>
+            </FlexCol>
+          </div>
+        )}
+        {isVideoOpen && (
+          <Modal
+            // eslint-disable-next-line @dailydotdev/daily-dev-eslint-rules/no-custom-color
+            className="px-8 bg-[black]"
+            kind={ModalKind.FlexibleCenter}
+            size={ModalSize.XLarge}
+            isOpen={isVideoOpen}
+            onRequestClose={onVideoModalClose}
+          >
+            <CloseButton
+              buttonSize={ButtonSize.Small}
+              className="top-3 right-3 text-white border-white !absolute !btn-secondary"
+              onClick={onVideoModalClose}
+            />
+            <iframe
+              className="w-full border-none aspect-video"
+              src="https://www.youtube.com/embed/igZCEr3HwCg"
+              title="YouTube video player"
+              allow="encrypted-media;web-share"
+              allowFullScreen
+            />
+          </Modal>
+        )}
+        <div
+          style={{
+            backgroundImage: `url(${cloudinary.referralCampaign.purpleEdgeGlow})`,
+          }}
+          className="hidden laptop:block absolute top-0 right-0 pointer-events-none h-[100vh] w-[29.375rem]"
+        />
+      </div>
+    </>
+  );
+};
+
+export const getServerSideProps: GetServerSideProps<PageProps> = async ({
+  query,
+  res,
+}) => {
+  const validateUserId = (value: string) => !!value && value !== '404';
+
+  const { userid: userId, cid } = query;
+  const campaign = Object.values(ReferralCampaignKey).find(
+    (item) => item === cid,
+  ) as ReferralCampaignKey;
+
+  if (!validateUserId(userId as string) || !campaign) {
+    return {
+      redirect: {
+        destination: '/',
+        permanent: false,
+      },
+    };
+  }
+
+  const result = await request<{ user: PageProps['referringUser'] }>(
+    graphqlUrl,
+    GET_REFERRING_USER_QUERY,
+    {
+      id: userId,
+    },
+  );
```

Comment:
Currently `request` will throw 500 and it will show our web app error page, I think it is valid because we want to see broken links or someone doing something fishy. We could redirect to homepage also as we do for not found campaigns eg. soft error when campaign expires. @idoshamun WDYT?

Reply:
Edge case. We can ignore for now